tutorials/flappy-bird/step_8/flappy-bird.cpp

```diff
@@ -0,0 +1,660 @@
+        auto new_pos_x = pos.x() - constants.scroll_speed * timer::time_step::get_fixed_delta_time();
+
+        // Set the new position value
+        registry.assign_or_replace<transform::position_2d>(pipe.body, new_pos_x, pos.y());
+
+        // Set cap position too
+        auto cap_pos = registry.get<transform::position_2d>(pipe.cap);
+        registry.assign_or_replace<transform::position_2d>(pipe.cap, new_pos_x, cap_pos.y());
+
+        // Return the info about if this pipe is out of the screen
+        return new_pos_x;
+    }
+};
+
+// Name this system
+REFL_AUTO (type(column_logic));
+
+// Player Logic System
+class player_logic final : public ecs::logic_update_system<player_logic> {
+public:
+    player_logic(entt::registry &registry, entt::entity player_) noexcept : system(registry), player(player_) {}
+
+    // Update, this will be called every tick
+    void update() noexcept final {
+        auto &registry = entity_registry_;
+
+        // Retrieve constants
+        const auto constants = registry.ctx<flappy_bird_constants>();
+
+        // Get current position of the pipe
+        auto pos = registry.get<transform::position_2d>(player);
+
+        // Add gravity to movement speed, multiply with delta time to apply it over time
+        movement_speed.set_y(movement_speed.y() + constants.gravity * timer::time_step::get_fixed_delta_time());
+
+        // Check if jump key is tapped
+        bool jump_key_tapped = input::virtual_input::is_tapped("jump");
+
+        // If jump is tapped, jump by adding jump force to the movement speed Y
+        if (jump_key_tapped) movement_speed.set_y(-constants.jump_force);
+
+        // Add movement speed to position to make the character move, but apply over time with delta time
+        pos += movement_speed * timer::time_step::get_fixed_delta_time();
+
+        // Do not let player to go out of the screen to top
+        if (pos.y() <= 0.f) {
+            pos.set_y(0.f);
+            movement_speed.set_y(0.f);
+        }
+
+        // Set the new position value
+        registry.assign_or_replace<transform::position_2d>(player, pos);
+
+        // ROTATION
+        // Retrieve props of the player
+        auto &props = registry.get<transform::properties>(player);
+
+        // Increase the rotation a little by applying delta time
+        float new_rotation = props.rotation + constants.rotate_speed * timer::time_step::get_fixed_delta_time();
+
+        // If jump button is tapped, reset rotation,
+        // If rotation is higher than the max angle, set it to max angle
+        if (jump_key_tapped)
+            new_rotation = 0.f;
+        else if (props.rotation > constants.max_angle)
+            new_rotation = constants.max_angle;
+
+        // Set the properties
+        registry.assign_or_replace<transform::properties>(player, transform::properties{.rotation = new_rotation});
+    }
+
+private:
+    entt::entity player;
```

Comment:
private member should have a `_` at the end according to our convention, please fix it everywhere